src/orchard/util/io.clj

```diff
@@ -22,3 +27,52 @@
             (System/setOut old-out))
           (when (= ps System/err) ;; `System/err` may have changed in the meantime (in face of concurrency)
             (System/setErr old-err)))))))
+
+(defn url-protocol
+  "Get the URL protocol as a string, e.g. http, file, jar."
+  [^java.net.URL url]
+  (.getProtocol url))
+
+(defn in-archive?
```

Comment:
`file-within-archive?` perhaps?

Originally I thought this meant a different thing ("is this an archive")

Reply:
Isn't this clear from the param? To me it seems obviously that we're checking if an URL is in archive.

Reply:
> To me it seems obviously that we're checking if an URL is in archive.

Hindsight is always 20/20 :) as mentioned at first the name genuinely confused me. If we pick this code up some years later we can forget these meanings, which is why being extra careful wrt clarity seems desirable for a project like this one.

No biggie though!

Reply:
This is now `url-to-file-within-archive?` as in "does this URL point at a file within an archive?"